packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -51,8 +57,42 @@ const defaultProps = {
 
 // eslint-disable-next-line react/prefer-stateless-function
 class DatePickerInput extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.handleOnButtonClick = this.handleOnButtonClick.bind(this);
+    this.handleOnKeyDown = this.handleOnKeyDown.bind(this);
+  }
+
+  handleOnButtonClick() {
+    // If the containing component (e.g. modal) has the focus trapped, the focus needs to be released so that the picker can get focus.
+    if (this.props.app && this.props.app.releaseFocus) {
+      this.props.app.releaseFocus();
```

Comment:
I was initially thinking of the other way around where the picker component would ask the containing component to "release" the focus. But I see your point for having the picker to "request" focus for itself.